File: app/src/main/java/com/reteno/sample/fragments/database/BaseReadDialogFragment.java
package com.reteno.sample.fragments.database;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

public abstract class BaseReadDialogFragment<Model, ViewHolderBinding extends ViewBinding, ViewHolder extends BaseReadViewHolder<Model, ViewHolderBinding>, Adapter extends BaseReadAdapter<Model, ViewHolderBinding, ViewHolder>> extends BaseDatabaseDialogFragment {

    protected com.reteno.sample.databinding.DialogDbReadBinding bindingMain;
    protected Adapter adapter;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        bindingMain = com.reteno.sample.databinding.DialogDbReadBinding.inflate(getLayoutInflater());
        return new AlertDialog.Builder(requireActivity())
                .setView(bindingMain.getRoot())
                .create();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        bindingMain = null;
    }

    @Override
    public void onStart() {
        super.onStart();
        initUi();
    }

    private void initUi() {
        initAdapter();
        initRecycler();
        initItems();
        initCount();
        initRemoveSection();
    }

    protected abstract void initAdapter();

    protected void initRecycler() {
        bindingMain.rvList.setAdapter(adapter);
    }

    protected abstract void initItems();

    protected abstract void initCount();

    private void initRemoveSection() {
        bindingMain.npRemoveEntries.setWrapSelectorWheel(false);
        bindingMain.npRemoveEntries.setMinValue(1);
        bindingMain.npRemoveEntries.setMaxValue(Integer.MAX_VALUE);
        bindingMain.btnRemoveEntries.setOnClickListener(v -> {
            int count = bindingMain.npRemoveEntries.getValue();
            deleteItems(count);

            initCount();
            initItems();
        });
    }

    protected abstract void deleteItems(int count);
}

File: app/src/main/java/com/reteno/sample/fragments/database/BaseReadDialogFragment.kt
package com.reteno.sample.fragments.database

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.View
import androidx.viewbinding.ViewBinding
import com.reteno.sample.databinding.DialogDbReadBinding

internal abstract class BaseReadDialogFragment<Model : Any, ViewHolderBinding : ViewBinding, ViewHolder : BaseReadViewHolder<Model, ViewHolderBinding>, Adapter : BaseReadAdapter<Model, ViewHolderBinding, ViewHolder>> :
    BaseDatabaseDialogFragment() {

    @JvmField
    protected var bindingMain: DialogDbReadBinding? = null

    @JvmField
    protected var adapter: Adapter? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        bindingMain = DialogDbReadBinding.inflate(layoutInflater)
        return AlertDialog.Builder(requireActivity())
            .setView(bindingMain!!.getRoot())
            .create()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        bindingMain = null
    }

    override fun onStart() {
        super.onStart()
        initUi()
    }

    private fun initUi() {
        initAdapter()
        initRecycler()
        initItems()
        initCount()
        initRemoveSection()
    }

    protected abstract fun initAdapter()
    protected fun initRecycler() {
        bindingMain!!.rvList.adapter = adapter
    }

    protected abstract fun initItems()
    protected abstract fun initCount()
    private fun initRemoveSection() {
        bindingMain!!.npRemoveEntries.setWrapSelectorWheel(false)
        bindingMain!!.npRemoveEntries.setMinValue(1)
        bindingMain!!.npRemoveEntries.setMaxValue(Int.MAX_VALUE)
        bindingMain!!.btnRemoveEntries.setOnClickListener { v: View? ->
            val count = bindingMain!!.npRemoveEntries.value
            deleteItems(count)
            initCount()
            initItems()
        }
    }

    protected abstract fun deleteItems(count: Int)
}
